app/extension/bitcoinInterface.js

```diff
@@ -0,0 +1,63 @@
+const bitcoin = require('../bitcoin/bitcoin.js');
+const persistence = require('../bitcoin/persistence.js');
+
+const address = new bitcoin.BitcoinAddress(persistence);
+let transfer = new bitcoin.BitcoinTransfer(address);
+
+
+function addDonation() {
+    if (!transfer.sufficientInput()) {  //If current input isn't enough, attempt to add more before proceeding
+        const unusedInputs = address.requestTransactionList()   //Get list of all unspent transactions associated with this address
+            .map(output => ({ tx_hash: output.tx_hash, tx_index: output.tx_index, value: output.value }))
+            .filter((output) => !transfer.inputs.some((input) => {  //Filter based on what ones are already in use for building the current transaction
+                if (input.index === output.tx_index && input.tx === output.tx_hash) {
+                    return input;
+                }
+                return undefined;
+            }));
+
+
+        do {
+            if (unusedInputs.length === 0) return false; //If we can't add enough input, return false (the current transaction should probably be sent)
```

Comment:
I think for MVP it is fine to drop the donation we are currently trying to add and just send off the transaction in its current state (since we know we have enough BC in the input to cover it)